lib/active_merchant/billing/gateways/pin.rb

```diff
@@ -178,13 +183,18 @@ def add_metadata(post, options)
       def add_3ds(post, options)
         if options[:three_d_secure]
           post[:three_d_secure] = {}
-          post[:three_d_secure][:version] = options[:three_d_secure][:version] if options[:three_d_secure][:version]
-          post[:three_d_secure][:eci] = options[:three_d_secure][:eci] if options[:three_d_secure][:eci]
-          post[:three_d_secure][:cavv] = options[:three_d_secure][:cavv] if options[:three_d_secure][:cavv]
-          post[:three_d_secure][:transaction_id] = options[:three_d_secure][:ds_transaction_id] || options[:three_d_secure][:xid]
+          if options[:three_d_secure][:enabled]
```

Comment:
There may be a situation where `options[:three_d_secure]` is not present so calling `options[:three_d_secure][:enabled]` would throw a `NoMethodError`. Could we add a guard at the top of this method to `return unless options[:three_d_secure]`?

Reply:
@aenand it's already guarded [here](https://github.com/activemerchant/active_merchant/pull/4720/files#diff-0e0df11ddf16e4e9c9d9225f2f82b44f460d63db9f13fe52b5324ff917f9248aR184).

Reply:
Oops Github's UI hid it from me. thanks!